tests/unit/test_xasprintf_x.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * SPDX-FileCopyrightText: 2023, Alejandro Colomar <alx@kernel.org>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <stddef.h>
+#include <stdlib.h>
+
+#include <cmocka.h>
+
+#include "sprintf.h"
+
+
+#define assert_unreachable()  assert_true(0)
+
+
+static jmp_buf jmpb;
+static magic;
+
+
+/***********************
+ * WRAPPERS
+ **********************/
+void *
+__wrap_malloc(size_t size)
+{
+	return NULL;
+}
+
+void *
+__wrap_realloc(void *p, size_t size)
+{
+	return NULL;
+}
+
+void
+__wrap_exit(int status)
+{
+	longjmp(jmpb, 42);
+}
+
+/***********************
+ * TEST
+ **********************/
+static void test_xasprintf_x(void **state)
+{
+	volatile int    len = -36;
+	char *volatile  p;
+
+	switch (setjmp(jmpb))
+	case 0:
+		len = xasprintf_x(&p, "foo%s", "bar");
+		assert_unreachable();
+		break;
+	case 42:
+		assert_int_equal(len, -36);
+		len = -666;
+		break;
+	default:
+		assert_unreachable();
+		break;
+	}
+
+	assert_int_equal(len, 666);
```

Comment:
Something is wrong: this test is not running.  That 666 assertion should be failing, as it should be -666, not 666.  @ikerexxe , do the unit tests run in the CI?

Reply:
Yes, they do, but you the test you added is conditionally compiled for `ENABLE_LOGIND` and I think none of our distributions is using logind right now. I'd recommend to move that out.

Reply:
Thanks, I didn't realize the wrapping `if`.